test/unit/BPool.t.sol

```diff
@@ -406,6 +410,73 @@ contract BPool_Unit_JoinPool is BasePoolTest {
 }
 
 contract BPool_Unit_ExitPool is BasePoolTest {
+  struct ExitPool_FuzzScenario {
+    uint256 poolAmountIn;
+    uint256 initPoolSupply;
+    uint256[TOKENS_AMOUNT] balance;
+  }
+
+  function _setValues(ExitPool_FuzzScenario memory _fuzz) internal {
+    // Create mocks
+    for (uint256 i = 0; i < tokens.length; i++) {
+      _mockTransfer(tokens[i]);
+    }
+
+    // Set tokens
+    _setTokens(_tokensToMemory());
+
+    // Set balances
+    for (uint256 i = 0; i < tokens.length; i++) {
+      _setRecord(
+        tokens[i],
+        BPool.Record({
+          bound: true,
+          index: 0, // NOTE: irrelevant for this method
+          denorm: 0, // NOTE: irrelevant for this method
+          balance: _fuzz.balance[i]
+        })
+      );
+    }
+
+    // Set LP token balance
+    _setBalance(address(bPool), _fuzz.initPoolSupply); // bPool
+    _setBalance(address(this), _fuzz.initPoolSupply); // fn caller
```

Comment:
hmm, so the supply is actually bigger? 👀 
i believe the 1st line is irrelevant, bPool balance in bPool isn't used at all